pyseir/inference/infer_rt.py

```diff
@@ -626,10 +712,266 @@ def infer_all(self, plot=True, shift_deaths=0):
+
+        plt.legend()
+        plt.savefig("scaledfig.pdf")
+
+        # Get features and labels
+        MIN_NUMBER_OF_DAYS = (
+            30  # I don't think it makes sense to predict anything until we have a month of data
+        )
+        PREDICT_DAYS = 5
+        # Create list of dataframes for testing
+        train_df_samples = self.create_df_list(train_set, MIN_NUMBER_OF_DAYS, PREDICT_DAYS)
+        X_train, Y_train = self.get_X_Y(
+            train_df_samples, PREDICT_DAYS, PREDICT_VARIABLE, MASK_VALUE
+        )
+
+        logging.info("done")
+        n_batch = 1
+        model, history = self.build_model(MASK_VALUE, 1, n_batch, 10, 0.01, X_train, Y_train)
+
+        logging.info("built model")
+
+        # Plot predictions for test and train sets
+        forecasts = list()
+        dates = list()
+        for i, j in zip(X_train, Y_train):
+            # original_df = self.get_reshaped_X(i, n_batch, X_scaler)
+            logging.info("df")
+            logging.info(i)
+            logging.info(i[0])
+            i = i.reshape(n_batch, i.shape[0], i.shape[1])
+            scaled_df = pd.DataFrame(np.squeeze(i))
+            logging.info("dataframe")
+            logging.info(scaled_df)
+            logging.info("dates")
+            logging.info(scaled_df[0])
+            thisforecast = scalers_dict[PREDICT_VARIABLE].inverse_transform(
+                model.predict(i, batch_size=n_batch)
+            )
+            logging.info("NATASHA THIS IS THE TYPE")
+            logging.info(type(model.predict(i, batch_size=n_batch)))
+            logging.info("appended forecast")
+            forecasts.append(thisforecast)
+            logging.info(scalers_dict[SIM_DATE_NAME])
+            logging.info("og dates")
+            # dates = scalers_dict[SIM_DATE_NAME].inverse_transform(scaled_df[0].to_numpy())
+            last_day = scaled_df.iloc[-1][0]
+            logging.info("last day")
+            # last_day_unscaled = scalers_dict[SIM_DATE_NAME].inverse_transform([[last_day]])
+            # logging.info('last day unscaled')
+            # logging.info(last_day_unscaled)
+            # predicted_days = np.arange(last_day, last_day + PREDICT_DAYS)
+            # dates.append(predicted_days)
+        logging.info("forecasts")
+        logging.info(forecasts)
+        logging.info("dates")
+        logging.info(dates)
+        # check if dictionary of scalers works
+
+        return
+
+    @staticmethod
+    def get_reshaped_X(input_X, n_batch, X_scaler):
+        i = input_X.reshape(n_batch, input_X.shape[0], input_X.shape[1])
+        output_df = pd.DataFrame(np.squeeze(input_X))
+        # original_df = pd.DataFrame(X_scaler.inverse_transform(output_df))
+        return original_df
+
+    @staticmethod
+    def build_model(
+        MASK_VALUE, epochs, n_batch, hidden_layer_dimensions, dropout, final_train_X, final_train_Y
+    ):
+        patience = 50
+        validation_split = 0.1
+        model = Sequential()
```

Comment:
Is there anyway we could separate the code for creating the model out from inside infer_rt? This file is getting pretty big